crates/cli-message/src/lib.rs

```diff
@@ -0,0 +1,104 @@
+extern crate proc_macro;
+
+use proc_macro::TokenStream;
+
+use proc_macro2::TokenStream as TokenStream2;
+use quote::quote;
+use syn::{parse_macro_input, Data, DeriveInput, Meta, Variant};
+
+mod error;
+mod parse;
+mod tok;
+
+use crate::parse::derive_message_from_im;
+use error::ParseMessageError;
+use parse::{parse_message_attribute, Message};
+use tok::build_match_arm_toks;
+
+fn impl_enum(
+    data: syn::DataEnum,
+    enum_name: &syn::Ident,
+) -> Result<TokenStream2, ParseMessageError> {
+    let mut variants_with_message: Vec<(&Variant, Message)> = Vec::new();
+
+    for variant in data.variants.iter() {
+        let message_attr = variant
+            .attrs
+            .iter()
+            .find_map(|attr| {
+                if let Meta::List(parts) = &attr.meta {
+                    if parts.path.is_ident("message") {
+                        return Some(parts);
+                    }
+                }
+                None
+            })
+            .ok_or(())
+            .map_err(|_| ParseMessageError::MissingMessageAttribute)?;
```

Comment:
No reason for `Err(())` step that I can see..?
```suggestion
            .ok_or(ParseMessageError::MissingMessageAttribute)?;
```